examples/Rules-method-nextBest-NextBestProbMtdMinDist.R

```diff
@@ -0,0 +1,43 @@
+# Example of usage for `NextBestProbMTDMinDist` NextBest class.
+
+# Create the data.
+my_data <- Data(
+  x = c(1.5, 1.5, 1.5, 2.5, 2.5, 2.5, 3.5, 3.5, 3.5),
+  y = c(0, 0, 0, 0, 0, 0, 1, 1, 0),
+  ID = 1:9,
+  cohort = c(1, 1, 1, 2, 2, 2, 3, 3, 3),
+  doseGrid = c(1.5, 2.5, 3.5, 4.5, 6, 7)
+)
+
+# Initialize the CRM model used to model the data.
+my_model <- my_model <- LogisticKadaneBetaGamma(
+  theta = 0.3,
+  xmin = 1.5,
+  xmax = 7,
+  alpha = 1,
+  beta = 19,
+  shape = 0.5625,
+  rate = 0.125
+)
+
+# Set-up some MCMC parameters and generate samples from the posterior.
+my_options <- McmcOptions(burnin = 100, step = 2, samples = 500)
+my_samples <- mcmc(my_data, my_model, my_options)
+
+# Define the rule for dose increments and calculate the maximum dose allowed.
+my_increments <- IncrementsNumDoseLevels(max_levels = 1)
+
+next_max_dose <- maxDose(my_increments, data = my_data)
+
+# Define the rule which will be used to select the next best dose
+# based on the 'NextBestProbMTDMinDist' class.
+prob_mtd_next_best <- NextBestProbMTDMinDist(target = 0.3)
+
+# Calculate the next best dose.
+dose_recommendation <- nextBest(
+  nextBest = prob_mtd_next_best,
+  doselimit = next_max_dose,
+  samples = my_samples,
+  model = my_model,
+  data = my_data
+)
```

Comment:
# Create the data with placebo
PL <- 0.001
data <- Data(x        = c(PL, 25, 25, 25, PL, 50, 50, 50, PL, 100, 100, 100),
             y        = c( 0,  0,  0,  0,  0,  0,  0,  0,  0,   0,    1,  0),
             cohort   = c( 1,  1,  1,  1,  2,  2,  2,  2,  3,   3,    3,  3),
             doseGrid = c(PL, seq(25, 300, 25)),
             ID       = 1:12,
             placebo  = TRUE
             )

# Initialize the CRM model used to model the data
model <- LogisticLogNormal(mean     = c(-0.85, 1),
                           cov      = matrix(c(1, -0.5, -0.5, 1),
                                            nrow=2
                                            ),
                           ref_dose = 100
                           )

# Set-up some MCMC parameters and generate samples from the posterior
options <- McmcOptions(burnin  = 1000,
                       step  = 2,
                       samples = 10000
                      )
set.seed(94)
samples <- mcmc(data, model, options)


# Define the rule for dose increments and calculate the maximum dose allowed
myIncrements <- IncrementsRelative(intervals  = c(0, 100, 200),
                                   increments = c(1, 0.5, 0.33)
)

nextMaxDose <- maxDose(myIncrements,
                       data = data
)

doselimit <- nextMaxDose

# Define the rule which will be used to select the next best dose
# based on the class 'NextBestProbMTDLTE'
mtdNextBest <- NextBestProbMTDLTE(target = 0.3,
                                  method = "max"
)

# Calculate the next best dose
doseRecommendation <- nextBest(mtdNextBest,
                               doselimit = nextMaxDose,
                               samples = samples,
                               model = model,
                               data = data
)